### DISCUSSION TITLE
Questions on building

### DISCUSSION THREAD

**Discussion by experk** [gerbera/gerbera#1133] [Q&A] [answered]
Hi there!
Me again =)

So, I've had few free hours and decided to again play around with gerbera on my router.
Previously I've tried the entware package and it did not work for my LG TV: no subtitles and mostly no video playback. Also, for whatever reason, I've got gerbera 1.7 out of package named 1.6.4.

There's a Debian Buster build available for my router, so I decided to give it a try.
Gerbera available from any Debian repo (even unstable) is severly outdated (1.1) so I didn't even try to use it.
Thus I've had no other choice then to try and build it from source and I've got A LOT of issues. As of now I was not able to successfully build it, but I do have few questions:

1. Why in release code was version set to 1.7 instead of 1.6.4? I mean here https://github.com/gerbera/gerbera/commit/2fa5af758ed840eb35d9e4dbfddb1a2ccacdd798 I guess this is why on Entware I've got gerbera telling me it's 1.7 even though in fact it was 1.6.4.
2. Are scripts from here https://github.com/gerbera/gerbera/tree/v1.6.4/scripts supposed to run automatically during build process? They did not run for me and I've spent quite some time building libupnp and taglib on my own, only to discover all those scripts after I've hit a wall of looking for lastfmlib.
3. Speaking of lastfmlib - if it's known to cause problems as per https://github.com/gerbera/gerbera/issues/995, maybe it's worth to remove "-DWITH_LASTFM=1" from https://github.com/gerbera/gerbera/blob/master/README.md? I mean, I didn't pay attention to it until cmake failed for whatever time it was.
4. I don't quite understand whether the cmake error about calling libupnp with invalid parameters can be ignored or is actually breaking my build attempt?

I know some questions will look dumb and maybe even ignorant, but I just hope that you guys are open to users with little to none coding background =)

**Comment by whyman** [comment, marked as answer]
1. That is an off-by-one error my part, the intention was to tag the commit previous (sigh)
2. The scripts are provided for convenience only, in more modern distros you can get all the libraries from the package trees.
3. Probably a good idea
4. That sounds like a build error to me, but would have to see the log.

Another option to avoid building could be to use the docker images if, you are so inclined.

**Reply by experk** [reply]
Thanks for a quick reply!
Understood, so in theory in a use case where I'm running Debian and options for modern packages are limited my building path should be something like:

1. Clone the 1.6.4 tag
2. Install prerequisites
3. Go to scripts directory and run those which I'm planning to build with
4. Run cmake and yada-yada-yada?

Regarding build error - I'm planning to nuke whatever I did yesterday and try a fresh build, so I can get you any logs you might wanna look at by the end of the day =)
Should I create an issue if I reproduce the error I'm having now?

I'd love to just go for containers, but considering that I'm already running a quite janky setup of chrooted debian on a mipsel router, I'm terrified to even imagine how many problems I'm going to have with docker =)

**Reply by whyman** [reply]
That sounds sensible.

If you get build errors do feel free to open an issue.

**Reply by experk** [reply]
Sharing here in case this feedback is useful - I did go through the steps I laid out above and I did get cmake to finish successfully, so using scripts definitely helped. However, as soon as I started make -j4, my router went brrrrrrrrrr and I've lost my Wi-Fi and wired network =)
The make was working though, judging by the disk activity, however, after an hour I decided to cancel the process.

So, now I'll either try running a QEMU mipsel VM on my desktop PC to build this or just stop trying to delay inevitable NAS purchase and succumb =)

Thank you for your time guys!

**Reply by KarlStraussberger** [reply]
You may try `make` without `-j4`. Running 4 compile jobs in parallel is too much for your small router.

**Reply by whyman** [reply]
Yes, try it with a -j1 to run a single compile job at once, for less resource usage
